### PR TITLE
Improve VB Audio test coverage from PR #11863

### PR DESCRIPTION
**Pull request by paul1956**
Part of Fixes in PR #11863

## Proposed changes
- Improve test coverage for VB Audio Tests

## Customer Impact
- Improves Code Coverage

## Regression? 
- No

## Risk
- None, Tests only

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12317)

### CODE REVIEW

File: src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs [hunk 1]
```diff
@@ -18,6 +18,37 @@ public void Play()
...
     }
 
+    [Theory]
+    [EnumData<AudioPlayMode>]
+    public void PlayAllModes_Throws(AudioPlayMode mode)
+    {
+        string location = Path.Combine(Path.GetTempPath(), GetUniqueName());
+        Audio audio = new();
+        Action testCode = () => audio.Play(location, mode);
+        testCode.Should().Throw<FileNotFoundException>();
```
Comment (Tanya-Solyanik) [inline src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs:28]: We are only validating file name for being null, FileNotFound is probably coming from System.Media, we shouldn't test for it.

File: src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs [hunk 2]
```diff
@@ -18,6 +18,37 @@ public void Play()
...
+    }
+
+    [Fact]
+    public void PlayBytes_Throws()
+    {
+        byte[] data = Array.Empty<byte>();
+        Audio audio = new();
+        Action testCode = () => audio.Play(data, AudioPlayMode.Background);
+        testCode.Should().Throw<InvalidOperationException>();
```
Comment (Tanya-Solyanik) [inline src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs:37]: We are not throwing InvalidOperationException, we shouldn't test for it.

File: src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs [hunk 3]
```diff
@@ -18,6 +18,37 @@ public void Play()
         testCode.Should().Throw<FileNotFoundException>();
     }
 
+    [Theory]
+    [EnumData<AudioPlayMode>]
```
Comment (Tanya-Solyanik) [inline src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs:22]: Is it possible to test an invalid value for AudioPlayMode?

File: src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs [hunk 4]
```diff
@@ -18,6 +18,37 @@ public void Play()
...
+        data = null;
+        testCode = () => audio.Play(data, AudioPlayMode.Background);
+        testCode.Should().Throw<ArgumentNullException>();
+    }
+
+    [Fact]
+    public void PlayEmptyFileName_Throws()
+    {
+        Audio audio = new();
+        Action testCode = () => audio.Play(string.Empty);
```
Comment (Tanya-Solyanik) [inline src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs:48]: Are you testing location set to `null` elsewhere? If not, then please add a case here.

File: src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs [hunk 5]
```diff
@@ -27,16 +58,17 @@ public void PlayInvalidMode_Throws()
...
+    [Fact]
+    public void PlayStream()
     {
-        string location = Path.Combine(Path.GetTempPath(), GetUniqueName());
+        var data = new MemoryStream();
         Audio audio = new();
-        Action testCode = () => audio.Play(location, mode);
-        testCode.Should().Throw<FileNotFoundException>();
+        Action testCode = () => audio.Play(data, AudioPlayMode.Background);
+        testCode.Should().Throw<InvalidOperationException>();
```
Comment (Tanya-Solyanik) [inline src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs:67]: We are not throwing this exception, System.Media should test it, we throw only nullReference and InvalidEnumArgument. Please add a test for invalid mode.

File: src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs (hunk 3 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs:22]: @Tanya-Solyanik I think @lonitra created a Theory Attribute to do this. I need to figure it out, elsewhere I used a really ugly solution in VB and it needs to be replaced also. I don't know C# well enough to use my VB hack.

Comment (Tanya-Solyanik) [reply src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs:22]: I would expect a new test like this one to work - 
[Theory]
[InvalidEnumData<AudioPlayMode>]
// test that your action throws appropriate exception.

Comment (paul1956) [reply src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/AudioTests.cs:22]: @Tanya-Solyanik added it and renamed and sorted tests so they are next to each other.

**Review by lonitra — APPROVED** [review]
LGTM, thank you

**Pull request merged**